### PR TITLE
Feature/reader stream refresh cards

### PR DESCRIPTION
**Pull request by nbradbury**
This is the first of several PRs which will address the reader stream refresh. Specific changes in this PR include:

* Post header now shows the author's avatar if available, otherwise it shows the blavatar if available, or nothing if neither are available
* Author + blog name are shown in the post header
* Removes the dateline that previously appeared below the title
* No tag
* Featured image height reduced to 100dp
* Adds a "Visit" icon to the post footer which opens the post in the internal browser
* Changes the color of the follow button to blue when the user isn't following the site

See #4776 and the before/after shots below for further details. Note that this, as well as subsequent related PRs, will undergo a design review after being merged into the feature branch.

before-and-after

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/models/ReaderPost.java [hunk 1]
```diff
@@ -205,6 +210,10 @@ public static ReaderPost fromJson(JSONObject json) {
         return post;
     }
 
+    public boolean hasImages() {
+        return hasText() && text.contains("<img ");
+    }
+
```
Comment (theck13) [inline WordPress/src/main/java/org/wordpress/android/models/ReaderPost.java]: I see you moved the `post.hasText() && post.text.contains("<img")` conditions to the new `hasImages()` method.  The second condition is changed from `"<img"` to `"<img "` in the new method.  I think using `"<img "` is a better check, but I just wanted to make sure that was intentional.

File: WordPress/src/main/res/color/reader_follow_button_text.xml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
 <selector xmlns:android="http://schemas.android.com/apk/res/android">
     <item android:color="@color/reader_following" android:state_selected="true" />
-    <item android:color="@color/reader_follow" />
+    <item android:color="@color/reader_follow_blue" />
```
Comment (theck13) [inline WordPress/src/main/res/color/reader_follow_button_text.xml]: It looks like the <b><i>Follow</i></b> button was changed from dark blue to light blue.  The dark blue matched the color of the <b><i>Follow</i></b> option in the menu.  Now they are different.  Show they be the same?  See the screenshots below for example.

follow

Comment (nbradbury) [inline WordPress/src/main/res/color/reader_follow_button_text.xml]: @theck13  I originally thought that menu icon should stay gray, but I've changed my mind. It's blue now, and I've removed the gray icon resource entirely. So, ready for another round!

**Comment by nbradbury** [comment]
> Is there a reason why the Follow button is in both the header and popup menu? It seems redundant. Is there a situation when the header isn't visible and the only way to follow/unfollow is through the popup menu?

The header used to be hidden for Discover, but it no longer is. Looking at the Calypso reader, I'm unclear what the rules are for showing the Follow button - it's there in some streams but not others. Mind if I address that in a separate PR so we can merge this one?

**Comment by theck13** [comment]
That's fine with me.  Reference that comment in the new pull request.

**Pull request merged**